apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.CommonTesting/SpannerTestDatabase.cs

```diff
@@ -79,25 +81,50 @@ private SpannerTestDatabase(string projectId)
             TestLogger.Install();
 
             ProjectId = projectId;
+            // Tests may connect to the emulator if the emulator host environment variable is set.
+            EmulatorDetection emulatorDetection = EmulatorDetection.EmulatorOrProduction;
+
             var builder = new SpannerConnectionStringBuilder
             {
                 Host = SpannerHost,
                 DataSource = $"projects/{ProjectId}/instances/{SpannerInstance}",
-                EmulatorDetection = EmulatorDetection.EmulatorOrProduction
+                EmulatorDetection = emulatorDetection
+            };
+
+            SessionPoolOptions options = new SessionPoolOptions();
+
+            // Create a builder that may create a client that connects to the
+            // emulator (if configured) so that we can configure the session
+            // pool. This is necessary because the emulator doesn't support
+            // running concurrent transactions so we need to configure the
+            // session pool to only have one active session at a time so that
+            // we don't prepare multiple concurrent transactions.
+            var clientBuilder = new SpannerClientBuilder
+            {
+                EmulatorDetection = emulatorDetection
             };
+            var emulatorBuilder = clientBuilder.MaybeCreateEmulatorClientBuilder();
```

Comment:
I wonder whether this should be in the production code, if everyone will always need to do that.
Do we know whether this is a long-term restriction?

Reply:
In general, the changes in this file (beyond just setting EmulatorDetection) feel like they're a symptom of us not doing quite enough to "just make it work".